### DISCUSSION TITLE
web-sys 0.4

### DISCUSSION THREAD

**Discussion by PaulWagener** [wasm-bindgen/wasm-bindgen#4090] [General]
As a foundational crate this project has a commendable dedication to semantic versioning and not breaking existing API's. With the web being a moving target some breaking changes are occasionally needed as demonstrated by the growing list of issues & PR's labeled 'breaking-change'. With no new version in sight it feels like any work done on breaking changes is currently a waste.

Is a 0.4 version even possible?
What blockers would need to be resolved before 0.4 could be released?
What should be included/excluded?
What do we want the update policy to be?

**Comment by daxpedda** [comment]
> Is a 0.4 version even possible?

Yes.
> What blockers would need to be resolved before 0.4 could be released?

I'm not aware of any.
> What should be included/excluded?

It would only be known if we have an overview on what potentially needs a breaking change in the first place.
> What do we want the update policy to be?

I'm unsure if we need one.
Updates are mostly stalled on a lack of maintainer resources, not on any design issues (at least not yet).

---

If we want to get the ball rolling on this, I would suggest somebody needs to sit down and collate all the breaking changes we would like to do and start a discussion for ones that need it.

See https://github.com/rustwasm/wasm-bindgen/issues/3111#issuecomment-2122251380.
